std-bits/base/src/main/java/org/enso/base/Random_Utils.java

```diff
@@ -4,6 +4,20 @@
 import java.util.Random;
 
 public class Random_Utils {
+  /** Static singleton Random instance. This is modified by Random.enso. */
+  private static Random randomSingleton = null;
+
+  public static Random getRandomSingleton() {
+    if (randomSingleton == null) {
+      seedRandomsSingleton(System.currentTimeMillis());
+    }
+    return randomSingleton;
+  }
+
+  public static void seedRandomsSingleton(long seed) {
+    randomSingleton = new Random(seed);
+  }
```

Comment:
`default_instance` isn't private and never will be. This is again a side-effect of the fact that we cannot create a true singleton instance in Enso. But yes I will document it there.

Reply:
> `default_instance` isn't private and never will be.

I'm not sure how this relates to the issue above, perhaps I did not understand what you meant?

> This is again a side-effect of the fact that we cannot create a true singleton instance in Enso

It's not that. The problem is you are not returning a singleton from within Java - you are actually overwriting the 'so-called' singleton upon `set_seed`. So we are not satisfying the 'singleton pattern contract' - since we are changing the instance.

Of course this is solvable (if we want this).

```java
class RandomInstanceHolder {
    private Random currentRandom = null;
    public Random getCurrentRandom() {
        if (currentRandom == null) { currentRandom = new Random(defaultSeed()); }
        return currentRandom;
    }

    private static int defaultSeed() { return System.getNanos something ... }
    private void setSeedInternal(int seed) {
        currentRandom = new Random(seed);
    }

    public RandomInstanceProvider SINGLETON = new RandomInstanceProvider();

    public setSeed(int seed) {
       SINGLETON.setSeed(seed);
    }
}
```

then in Enso we can do:
```
type Random_Generator
    Value java_random_holder

    default_generator = Random_Generator.Value RandomInstanceHolder.SINGLETON

    java_random self = self.java_random_holder.getCurrentRandom
```

now every instance method we define on `Random_Generator` that will rely on `self.java_random`, will use the _currently up-to date_ seed, regardless of when the instance was instantiated.

Reply:
Of course what I present is one solution. Alternative, maybe more elegant, although also requiring a bit more work is to make our own `Random` interface that allows changing of the seed and exposes similar methods to java random (possibly by delegating to it).

My point is - if we want to ensure that the seed is updated globally - we can definitely do it.

Reply:
Done.